src/mmdata.c

```diff
@@ -2413,16 +2464,48 @@ long g_pntrStartTempAllocStack = 0;   /* Where to start freeing temporary alloca
                                     when pntrLet() is called (normally 0, except in
                                     special nested vstring functions) */
 /*!
- * \var pntrTempAllocStack
- * \brief a \ref stack "stack" of \a temp_pntrString.
+ * \var pntrString *pntrTempAllocStack[]
+ * \brief a \ref stack "stack" of \ref temp_pntrString.
  *
- * Holds pointers to temporarily allocated data of type \a pntrString.  Such a
- * \ref stack "stack" contains strictly __local__ data of a function, not
- * accessed from outer levels.
+ * Holds pointers to temporarily allocated data of type \ref temp_pntrString.  Such
+ * a \ref stack "stack" is primarily designed to operate like one for temporary
+ * allocated ad hoc operands, as described in \ref stack.  The stack top index
+ * is \ref g_pntrTempAllocStackTop, always refering to the next push position.
+ * The \ref g_pntrStartTempAllocStack supports nested operations by indicating
+ * where the operands for the upcoming operation start from.
+ * \attention A \ref pntrString consists of an array of pointers.  These
+ *   pointers may themself refer data that needs a clean up, when the last
+ *   reference  to it disappears (such as deallocating memory for example).
+ *   There is no automatic procedure handling such cases when pointers are
+ *   popped off the stack to be freed.
+ * \bug The element type should be temp_pntrString, because a NULL_PNTRSTRING
+ *   must not be pushed on the stack.
```

Comment:
I try to keep the code as it is as strictly as possible.  This is because many rely on it to a great extent.  Its easy to accept pure documentation changes when you can rest assured that nothing that really matters to the machine is touched.

Then I have a particular problem: I work my way bottom-up.  That way I miss the bird's eye view, and this limitation prevents me from doing anything on the fly.  An improvement in one spot may easily break something in a remote location.

Believe me I have tons of ideas on how to refactor code.